Detectors/ITSMFT/ITS/RawPixelReaderWorkflow/src/RawPixelReaderSpec.cxx

```diff
@@ -0,0 +1,608 @@
+#include <vector>
+#include <TTree.h>
+#include <TFile.h>
+#include <TStopwatch.h>
+#include <string>
+#include "TTree.h"
+
+#include "Framework/ControlService.h"
+#include "ITSMFTReconstruction/ChipMappingITS.h"
+#include "ITSMFTReconstruction/GBTWord.h"
+#include "ITSMFTReconstruction/PayLoadCont.h"
+#include "ITSMFTReconstruction/PixelData.h"
+#include "DataFormatsITSMFT/ROFRecord.h"
+#include "ITSMFTReconstruction/RawPixelReader.h"
+#include "CommonDataFormat/InteractionRecord.h"
+#include "ITSRawWorkflow/RawPixelReaderSpec.h"
+#include "DetectorsBase/GeometryManager.h"
+#include <TCanvas.h>
+#include <iostream>
+#include <dirent.h>
+#include <stdio.h>
+#include <algorithm>
+#include <iterator>
+#include <thread>
+
+using namespace o2::framework;
+using namespace o2::itsmft;
+using namespace std;
+
+namespace o2
+{
+namespace its
+{
+
+void RawPixelReader::init(InitContext& ic)
+{
+  IndexPush = 0;
+
+  std::ifstream RunFileType("Config/RunType.dat");
+  RunFileType >> RunType;
+
+  LOG(DEBUG) << "OLD CONFIG: RunFileType = " << RunType;
+
+  if (RunType == 0) {
+    std::ifstream EventPush("Config/ConfigFakeRate.dat");
+    EventPush >> EventPerPush >> TrackError >> workdir;
+  }
+
+  if (RunType == 1) {
+    std::ifstream EventPush("Config/ConfigThreshold.dat");
+    EventPush >> EventPerPush >> TrackError >> workdir;
+  }
+
+  LOG(DEBUG) << "OLD CONFIG: EventPerPush = " << EventPerPush << "   TrackError = " << TrackError << "  workdir = " << workdir;
+  LOG(DEBUG) << "DONE Reset Histogram Decision";
+
+  o2::base::GeometryManager::loadGeometry();
```

Comment:
Why do you load geometry if you never use it?

Reply:
Suggestions taken!